app/scripts/PixiTrack.js

```diff
@@ -294,15 +296,43 @@ class PixiTrack extends Track {
     );
   }
 
+  /** Set an error for this track.
+   *
+   * The error can be associated with a source so that multiple
+   * components within the track can set their own independent errors
+   * that will be displayed to the user without overlapping.
+   *
+   * @param {string} error The error text
+   * @param {string} source The source of the error
+   */
+  setError(error, source) {
+    this.errorTexts[source] = error;
+
+    this.drawError();
+  }
+
   drawError() {
     this.errorText.x = this.position[0] + this.dimensions[0] / 2;
     this.errorText.y = this.position[1] + this.dimensions[1] / 2;
 
-    this.errorText.text = this.errorTextText;
+    let errorTextText = '';
```

Comment:
Small nit, took me a little bit to see what was going on here (filter + reduce). 

Not essential, but maybe we could make that more explicit. E.g.,

```javascript
const errorTextText = Object.values(this.errorTexts)
  .filter(x => x && x.length)
  .reduce((acc, x) => acc ? `${acc}\n${x}` : x, '');
```

Reply:
You kids and your idiomatic expressions!